### PR TITLE
Woptim/update shared ci

### PR DESCRIPTION
**Pull request by adrienbernede**
### CI Updates

This PR supersedes #1384 

We perform the following updates:

-   spack@0.18.1 -> spack@0.19.0.
-   radiuss-spack-configs updated accordingly to support new Spack syntax.
-   radiuss-shared-ci@2022.09.0 -> [radiuss-shared-ci@2022.12.0rc](mailto:radiuss-shared-ci@2022.12.0rc).

In particular, radiuss shared ci update brings:

-   support for spack@0.19.1
-   improved performance (not more project clone in status jobs)
-   improved logs (prints a complete reproducer of the current job)
-   jobs will not be triggered on a machine that is not available
-   possibility to prevent CI to run on Draft PRs (activated).

### CODE REVIEW

File: .gitlab/lassen-build-and-test-extra.yml
Comment (davidbeckingsale) [inline .gitlab/lassen-build-and-test-extra.yml:139]: Mismatched "?

**Comment by rhornung67** [comment]
@adrienbernede we can merge this after we coordinate a few other things. I'll let you know when we can merge it. Hopefully, it will be in a day or two.

**Comment by adrienbernede** [comment]
@rhornung67 before merging I just need to release radiuss-shared-ci and merge radiuss-spack-configs changes in the main branch. (which means I then need to update the references here). I’ll try to get that done today.

**Comment by adrienbernede** [comment]
@rhornung67 I did the above. Ready to merge provided that tests pass.

**Comment by adrienbernede** [comment]
@rhornung67 those tests really tend to fail in a non-reproducible way:
```
	 78 - test-forall-segment-multiple-ReduceMinLoc-Hip.exe (Failed)
	 79 - test-forall-segment-multiple-ReduceMaxLoc-Hip.exe (Failed)
```

**Comment by rhornung67** [comment]
Yes, I know. We don't understand it.

**Comment by rhornung67** [comment]
@davidbeckingsale I'm good with this. Do you want to review before merging?

**Review by rhornung67 — APPROVED** [review]
Thank you  @adrienbernede This looks good.

**Review by davidbeckingsale — APPROVED** [review]

**Comment by adrienbernede** [comment]
Thank you @davidbeckingsale @rhornung67!
FYI, I’m not authorized to hit the merge button.

**Comment by rhornung67** [comment]
@adrienbernede @davidbeckingsale I will merge this later today.

**Comment by rhornung67** [comment]
@davidbeckingsale are you ok with merging this? Adrien is on vacation for the next two weeks.

**Comment by davidbeckingsale** [comment]
Yes this looks good to me.

**Pull request merged**
